include/mdw_dslash5_tensor_core.cuh

```diff
@@ -283,23 +287,19 @@ namespace quda
         __float_max_abs_floats__(warp_max, v(spin, color).imag());
       }
     }
-    // Find the maximum absolute value in a warp across different lanes
-    warp_wise_reduce_float(warp_max);
-    // Now lane 0 of each warp holds the maximum value
-    if (lane_id == 0) { smem_scale[warp_id] = warp_max; }
-    __syncthreads();
-    if (warp_id == 0) {
-      warp_max = (lane_id < ((blockDim.x * blockDim.y) >> 5)) ? smem_scale[lane_id] : 0.0f;
-      warp_wise_reduce_float(warp_max);
-      if (lane_id == 0) { smem_scale[0] = warp_max / target_scale; }
-    }
+
+    typedef cub::BlockReduce<float, block_x, cub::BLOCK_REDUCE_WARP_REDUCTIONS, block_y, 1> BlockReduce;
+    __shared__ typename BlockReduce::TempStorage temp_storage;
+    float aggregate = BlockReduce(temp_storage).Reduce(warp_max, cub::Max());
+
+    if (warp_id == 0 && lane_id == 0) { smem_scale[0] = aggregate / target_scale; }
```

Comment:
Curious: any perf different from using CUB for the reduction here?

Reply:
There is visible difference in performance.

Reply:
CUB is faster I assume?

Reply:
Sorry previously I forget to type the "no" ... there is *no* visible difference